e2e/test/ASCforIoTOmsClient.cs

```diff
@@ -0,0 +1,102 @@
+        private readonly string _appId = Configuration.Oms.AadAppId;
+        private readonly string _appKey = Configuration.Oms.AadAppKey;
+
+        private readonly HttpClient _client;
+        private readonly AuthenticationContext _authenticationContext;
+        private readonly string _queryUri;
+
+        public static ASCforIoTOmsClient CreateClient()
+        {
+            return new ASCforIoTOmsClient();
+        }
+
+        private ASCforIoTOmsClient()
+        {
+            _client = new HttpClient();
+            _authenticationContext = new AuthenticationContext("https://login.windows.net/" + _aadTenant);
+            _queryUri = string.Format(CultureInfo.InvariantCulture, QueryUriTemplate, LogAnalyticsApiVersion, _workspaceId);
+        }
+
+        public async Task<bool> IsRawEventExist(string deviceId, string eventId)
+        {
+            bool isEventExist = false;
+            string query = string.Format(CultureInfo.InvariantCulture, RawEventQueryTemplate, deviceId, eventId);
+            var sw = new Stopwatch();
+            sw.Start();
+            while (!isEventExist && sw.Elapsed.TotalMinutes < 30)
+            {
+                isEventExist = await DoQuery(query).ConfigureAwait(false);
+                await Task.Delay(TimeSpan.FromMinutes(1)).ConfigureAwait(false);
+            }
+
+            sw.Stop();
+            return isEventExist;
+        }
+
+        private async Task<bool> DoQuery(string query)
+        {
+            using (HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Post, _queryUri))
+            {
+                string accessToken = await GetAccessToken().ConfigureAwait(false);
+                request.Headers.Authorization = new AuthenticationHeaderValue("Bearer", accessToken);
+                request.Content = CreateRequestContent(query);
+                using (HttpResponseMessage response = await _client.SendAsync(request).ConfigureAwait(false))
+                {
+                    string responseAsString = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
+                    JObject responseBody = JObject.Parse(responseAsString);
+
+                    return responseBody["tables"][0]["rows"].HasValues;
+                }
+            }
+        }
+
+        private async Task<string> GetAccessToken()
+        {
+            ClientCredential cc = new ClientCredential(_appId, _appKey);
+            AuthenticationResult result = await _authenticationContext.AcquireTokenAsync(Audience, cc).ConfigureAwait(false);
+
+            return result.AccessToken;
+        }
+
+        private StringContent CreateRequestContent(string query)
+        {
+            JObject body = new JObject();
+            body["query"] = query;
+            body["timespan"] = TimeSpan.FromHours(2);
+
+            StringContent content = new StringContent(JsonConvert.SerializeObject(body));
+            content.Headers.ContentType = new MediaTypeHeaderValue("application/json");
```

Comment:
There must be a const for the json mime type

Reply:
https://stackoverflow.com/questions/20655938/application-json-constant-in-net-framework